main/streams/cast.c

```diff
@@ -187,6 +187,12 @@ void php_stream_mode_sanitize_fdopen_fopencookie(php_stream *stream, char *resul
 }
 /* }}} */
 
+static inline bool is_stream_user_stream(const php_stream *stream)
+{
+	return (strlen(stream->ops->label) == strlen("user-space"))
+		&& strncmp(stream->ops->label, "user-space", strlen("user-space")) == 0;
```

Comment:
An other way to do that would be to compare stream->ops to user_stream_wops. We do something like this in https://github.com/php/php-src/blob/ebf26af71a6e9eb800f04d1498540d187077a021/main/streams/xp_socket.c#L655